isolator/interface.hpp

```diff
@@ -0,0 +1,6 @@
+#ifndef __INTERFACE_HPP__
+#define __INTERFACE_HPP__
+
+#include <isolator/interface.pb.h>
```

Comment:
I'm not sure I follow - I'm building this from the Dockerfile, so my compile command is:
```
RUN ./bootstrap && \
    rm -rf build && \
    mkdir build && \
    cd build && \
    export LD_LIBRARY_PATH=LD_LIBRARY_PATH:/usr/local/lib && \
    ../configure --with-mesos-build-dir=/mesos/build --with-mesos-root=/mesos && \
    make all
```
I can modify the Dockerfile to add the `--with-mesos`, but what would the `/path/to/mesos/installation` be for the `mesosphere/mesos-modules-dev-phusion
` image?

Here's my build output dump:
```
Step 11 : RUN ./bootstrap &&     rm -rf build &&     mkdir build &&     cd build &&     export LD_LIBRARY_PATH=LD_LIBRARY_PATH:/usr/local/lib &&     ../configure --with-mesos-build-dir=/mesos/build --with-mesos-root=/mesos &&     make all
 ---> Running in e4f7046cb899
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force --warnings=all -I m4
autoreconf: configure.ac: tracing
configure.ac:17: warning: back quotes and double quotes must not be escaped in: unrecognized option: $[1]
configure.ac:17: Try \`$[0] --help' for more information.
aclocal.m4:625: LT_OUTPUT is expanded from...
configure.ac:17: the top level
configure.ac:17: warning: back quotes and double quotes must not be escaped in: unrecognized argument: $[1]
configure.ac:17: Try \`$[0] --help' for more information.
aclocal.m4:625: LT_OUTPUT is expanded from...
configure.ac:17: the top level
autoreconf: running: libtoolize --copy --force
libtoolize: putting auxiliary files in `.'.
libtoolize: copying file `./ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIR, `m4'.
libtoolize: copying file `m4/libtool.m4'
libtoolize: copying file `m4/ltoptions.m4'
libtoolize: copying file `m4/ltsugar.m4'
libtoolize: copying file `m4/ltversion.m4'
libtoolize: copying file `m4/lt~obsolete.m4'
autoreconf: running: /usr/bin/autoconf --force --warnings=all
configure.ac:17: warning: back quotes and double quotes must not be escaped in: unrecognized option: $[1]
configure.ac:17: Try \`$[0] --help' for more information.
m4/libtool.m4:609: LT_OUTPUT is expanded from...
configure.ac:17: the top level
configure.ac:17: warning: back quotes and double quotes must not be escaped in: unrecognized argument: $[1]
configure.ac:17: Try \`$[0] --help' for more information.
m4/libtool.m4:609: LT_OUTPUT is expanded from...
configure.ac:17: the top level
autoreconf: configure.ac: not using Autoheader
autoreconf: running: automake --add-missing --copy --force-missing --warnings=all
configure.ac:5: installing './compile'
configure.ac:15: installing './config.guess'
configure.ac:15: installing './config.sub'
configure.ac:10: installing './install-sh'
configure.ac:10: installing './missing'
Makefile.am: installing './depcomp'
autoreconf: Leaving directory `.'
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking whether gcc and cc understand -c and -o together... yes
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
/bin/bash: /root/missing: No such file or directory
configure: WARNING: 'missing' script is too old or missing
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... yes
checking for style of include used by make... GNU
checking whether make supports nested variables... yes
checking dependency style of gcc... gcc3
checking dependency style of g++... gcc3
checking for python... /usr/bin/python
checking for a version of Python >= '2.1.0'... yes
checking for the distutils Python package... yes
checking for Python include path... -I/usr/include/python2.7
checking for Python library path... -L/usr/lib -lpython2.7
checking for Python site-packages path... /usr/lib/python2.7/dist-packages
checking python extra libraries... -lpthread -ldl  -lutil -lm
checking python extra linking flags... -Xlinker -export-dynamic -Wl,-O1 -Wl,-Bsymbolic-functions
checking consistency of all components of python development environment... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking how to print strings... printf
checking for a sed that does not truncate output... /bin/sed
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for fgrep... /bin/grep -F
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for mt... mt
checking if mt is a manifest tool... no
checking how to run the C preprocessor... gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... /usr/bin/ld -m elf_x86_64
checking if the linker (/usr/bin/ld -m elf_x86_64) is GNU ld... yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC -DPIC
checking if g++ PIC flag -fPIC -DPIC works... yes
checking if g++ static flag -static works... yes
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
configure: creating ./config.lt
config.lt: creating libtool
checking if compiling with clang... no
configure: GCC version: 4.8
configure: Setting up CXXFLAGS for g++ version >= 4.8
checking whether g++ supports C++11 features by default... no
checking whether g++ supports C++11 features with -std=c++11... yes
checking glog/logging.h usability... yes
checking glog/logging.h presence... yes
checking for glog/logging.h... yes
checking picojson.h usability... yes
checking picojson.h presence... yes
checking for picojson.h... yes
checking boost/lexical_cast.hpp usability... yes
checking boost/lexical_cast.hpp presence... yes
checking for boost/lexical_cast.hpp... yes
checking boost/functional/hash.hpp usability... yes
checking boost/functional/hash.hpp presence... yes
checking for boost/functional/hash.hpp... yes
checking google/protobuf/message.h usability... yes
checking google/protobuf/message.h presence... yes
checking for google/protobuf/message.h... yes
checking for protoc... protoc
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating isolator/modules.json
config.status: executing depfiles commands
config.status: executing libtool commands
depbase=`echo isolator/network_isolator.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
	/bin/bash ./libtool  --tag=CXX   --mode=compile g++ -DPACKAGE_NAME=\"metaswitch-modules\" -DPACKAGE_TARNAME=\"metaswitch-modules\" -DPACKAGE_VERSION=\"0.1\" -DPACKAGE_STRING=\"metaswitch-modules\ 0.1\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DPACKAGE=\"metaswitch-modules\" -DVERSION=\"0.1\" -DHAVE_PYTHON=\"2.7\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_BOOST_LEXICAL_CAST_HPP=1 -DHAVE_BOOST_FUNCTIONAL_HASH_HPP=1 -I. -I..  -I/mesos/include -I/mesos/3rdparty/libprocess/include -I/mesos/3rdparty/libprocess/3rdparty/stout/include -I/mesos/build/include -I../include -Wall -Werror   -g -O2 -Wno-unused-local-typedefs -std=c++11 -MT isolator/network_isolator.lo -MD -MP -MF $depbase.Tpo -c -o isolator/network_isolator.lo ../isolator/network_isolator.cpp &&\
	mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  g++ -DPACKAGE_NAME=\"metaswitch-modules\" -DPACKAGE_TARNAME=\"metaswitch-modules\" -DPACKAGE_VERSION=\"0.1\" "-DPACKAGE_STRING=\"metaswitch-modules 0.1\"" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DPACKAGE=\"metaswitch-modules\" -DVERSION=\"0.1\" -DHAVE_PYTHON=\"2.7\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_BOOST_LEXICAL_CAST_HPP=1 -DHAVE_BOOST_FUNCTIONAL_HASH_HPP=1 -I. -I.. -I/mesos/include -I/mesos/3rdparty/libprocess/include -I/mesos/3rdparty/libprocess/3rdparty/stout/include -I/mesos/build/include -I../include -Wall -Werror -g -O2 -Wno-unused-local-typedefs -std=c++11 -MT isolator/network_isolator.lo -MD -MP -MF isolator/.deps/network_isolator.Tpo -c ../isolator/network_isolator.cpp  -fPIC -DPIC -o isolator/.libs/network_isolator.o
In file included from ../isolator/network_isolator.cpp:66:0:
../isolator/interface.hpp:4:35: fatal error: isolator/interface.pb.h: No such file or directory
 #include <isolator/interface.pb.h>
                                   ^
compilation terminated.
make: *** [isolator/network_isolator.lo] Error 1
The command '/bin/sh -c ./bootstrap &&     rm -rf build &&     mkdir build &&     cd build &&     export LD_LIBRARY_PATH=LD_LIBRARY_PATH:/usr/local/lib &&     ../configure --with-mesos-build-dir=/mesos/build --with-mesos-root=/mesos &&     make all' returned a non-zero code: 2
make: *** [images] Error 1
```